common: bump systemReserved to 4 CPU and 4Gi RAM

jtriley opened the pull request:
We're currently debugging an issue with timeouts/unavailability of at least the authentication cluster operator and potentially others (e.g. apiserver/kube-apiserver) on the nerc-ocp-infra cluster. In the case we opened with RedHat, they pointed out the warning about reserved system memory. This bumps the system memory from the default of 500m CPU and 1Gi RAM, via a kubeletconfig in the common overlay, to see if this helps given the number of pods running on this cluster.

See:
- https://github.com/OCP-on-NERC/operations/issues/127
- https://access.redhat.com/support/cases/#/case/03513513

jtriley committed:
common: bump systemReserved to 4 CPU and 4Gi RAM

We're currently debugging an issue with timeouts/unavailability of
at least the authentication cluster operator and potentially others
(e.g. apiserver/kube-apiserver). In the case we opened with RedHat, they
pointed out the warning about reserved system memory. This bumps the
system memory from the default of 500m CPU and 1Gi RAM to see if this
helps given the number of pods running on this cluster.

See:
- https://github.com/OCP-on-NERC/operations/issues/127
- https://access.redhat.com/support/cases/#/case/03513513

larsks reviewed (approved):
Should we move this into the common overlay instead (so that it also applies to the prod cluster)?

jtriley committed:
prod: include worker nodes in systemReserved

The prod cluster has worker nodes which can also host internal service
pods. This patches the system-reserved kubeletconfig to include worker
nodes in the mcp label selector.
